frontend/src/templates/Field/Checkbox/CheckboxField.tsx

```diff
@@ -34,43 +38,35 @@ export const CheckboxField = ({
     () => `${schema._id}.${CHECKBOX_OTHERS_INPUT_KEY}`,
     [schema._id],
   )
-  const checkboxInputName = `${schema._id}.value`
+  const checkboxInputName = useMemo(() => `${schema._id}.value`, [schema._id])
 
-  const validationRules = useMemo(() => {
-    return {
-      ...createCheckboxValidationRules(schema),
-      deps: [othersInputName],
-    }
-  }, [othersInputName, schema])
-
-  const {
-    watch,
-    register,
-    formState: { isValid, isSubmitting, errors },
-  } = useFormContext()
-  const checkboxValues = watch(checkboxInputName, [])
-  const isOthersSelected = useMemo(
-    () =>
-      Array.isArray(checkboxValues) &&
-      checkboxValues.includes(CHECKBOX_OTHERS_INPUT_VALUE),
-    [checkboxValues],
+  const validationRules = useMemo(
+    () => createCheckboxValidationRules(schema),
+    [schema],
   )
 
+  const { register, getValues, trigger } = useFormContext()
+  const { isValid, isSubmitting, errors } = useFormState({
+    name: schema._id,
+  })
+
   const othersValidationRules = useMemo(
     () => ({
       validate: (value: string) => {
+        const currCheckedVals = getValues(checkboxInputName)
         return (
-          !isOthersSelected ||
+          !(
+            Array.isArray(currCheckedVals) &&
+            currCheckedVals.includes(CHECKBOX_OTHERS_INPUT_VALUE)
+          ) ||
           !!value ||
           'Please specify a value for the "others" option'
         )
       },
     }),
-    [isOthersSelected],
+    [checkboxInputName, getValues],
   )
 
-  const othersInputError: FieldError | undefined = get(errors, othersInputName)
```

Comment:
How does removing this reference help here?

Reply:
As discussed, it does nothing ><. Just moved it in since we get nothing from declaring it outside